#171784086 Reset Password via email

Francois-MUGOROZI opened the pull request:
#### What does this PR do?
Enable user to reset a password when forgotten
#### Description of Task to be completed?
- Create a forgot password endpoint and send reset link to user's email
- Create reset password endpoint to change user password with new one
#### How should this be manually tested?
Give that the user has signed up (refer to signup feature); clone the repo
cd into it and run npm install then npm run dev.
using postman, send forgot password at localhost:5000/api/auth/forgotpassword with email in the body.
Refer to the API docs by browsing: http://localhost:5000/api-docs/
If email is yours, you will get a link in your inbox

#### What are the relevant pivotal tracker stories?
[171784086](https://www.pivotaltracker.com/n/projects/2438874/stories/171784086)
#### Screenshots (if appropriate)
![Capture](https://user-images.githubusercontent.com/53614123/78048105-ba038b00-7325-11ea-83bd-225bc2f0af1a.JPG)
![Capture](https://user-images.githubusercontent.com/53614123/78246312-6a43d180-7495-11ea-8add-96a5b2248b2d.JPG)

benshidanny11 reviewed (dismissed):
- Good job @Francois-MUGOROZI , 
**Feedbacks**
- I think you are importing files that don't exist in project. 
      Ex: import userService from '../services/userService'; & import sendEmail from '../helpers/email.create'; 
- In model directory of sequelize,  I think is where all models should be, example : User,...

irfiacre commented on server/controllers/usercontroller.js:
```diff
@@ -18,46 +19,98 @@ class User {
+      return util.send(res);
+    }
+
+    // generate token and send it to email
```
Remove the comments from the code

irfiacre reviewed (changes requested):
Nice work @Francois-MUGOROZI.  Just Resolve The hound errors and make sure the tests are running.

kabundege reviewed (approved):
-  A tip on hound errors fix 
 ```just add a comma on the last element in a mentioned objects, it will be solved.```
 - You should squash our commits and remain with one.
 - make sure your tests are running

Francois-MUGOROZI committed:
ft(reset password): add reset password via email link

 - added forgot password endpoint
 - addeed reset password endpoint
 - added tests
 - implement feedback
[Finish #171784086]

bdushimi commented on server/controllers/usercontroller.js:
```diff
@@ -18,48 +19,72 @@ class User {
+    return util.send(res);
+  }
+
+  static async forgotPassword(req, res) {
```
@Francois-MUGOROZI sendResetPasswordEmail is more appropriate as the function name.

bdushimi commented on server/controllers/usercontroller.js:
```diff
@@ -18,48 +19,72 @@ class User {
+  static async forgotPassword(req, res) {
+    const { email } = req.body;
+    const resetToken = encode({ email });
+    const restUrl = `${process.env.RESET_URL}?token=${resetToken}`;
```
@Francois-MUGOROZI maybe you can change this variable name too to a name that describes what it contains

bdushimi commented on server/controllers/usercontroller.js:
```diff
@@ -18,48 +19,72 @@ class User {
+    const footer = `For more info or questions, contact : ${process.env.EMAIL}`;
+    const html = `<div>${message}<a href="${restUrl}" style="background-color:#028b95;color:white;padding:10px 20px;text-decoration:none">Clik Here</a><br/><br/>${footer}</div>`;
+
+    sendEmail(html, subj, email);
```
@Francois-MUGOROZI may you handle the error just in case nodemailer fails to send the email.

Francois-MUGOROZI committed:
ft(reset-password): add reset password via email
- implemented feedback and refactor
[Finish]
